src/core/thread/mle_router.cpp

```diff
@@ -2842,6 +2847,17 @@ void MleRouter::HandleDiscoveryRequest(const Message &aMessage, const Ip6::Messa
 
     error = SendDiscoveryResponse(aMessageInfo.GetPeerAddr(), aMessage.GetPanId());
 
+    if (mDiscoveryRequestCallback != nullptr && hasDiscoveryRequest)
```

Comment:
Can we give `mDiscoveryRequestCallback` a chance to determine how the node will handle the `DiscoveryRequest` ? For example, if `mDiscoveryRequestCallback` returns `false`, we don't `SendDiscoveryResponse`.

Reply:
I also considered this, but I don't have a use case of it for now. I prefer not adding a feature before we are going to use it, otherworldly it only increases maintenance burden and decreases code coverage. Thoughts?